neo/Network/P2P/ProtocolHandler.cs

```diff
@@ -156,7 +156,8 @@ private void OnGetBlocksMessageReceived(GetBlocksPayload payload)
             BlockState state = Blockchain.Singleton.Store.GetBlocks().TryGet(hash);
             if (state == null) return;
             List<UInt256> hashes = new List<UInt256>();
-            for (uint i = 1; i <= InvPayload.MaxHashesCount; i++)
+            hashes.Add(hash);
```

Comment:
I think the code that requests always gives the hash before the cone that it actually expects to get back first

Reply:
I did not understand what you mean, which lines?

Reply:
I think not, @jsolman,  this is exactly the piece of code of the #495 and the function that is being fixed there, take another look if you can.

The point is that all this logic looks kind of reversed.

I think that we should ask for the block that we really need and a more precise limit, instead of 500 next blocks (some of then may had already been received and are on GlobalTasks in order to be processed and asked for data).

Reply:
Looks like you are right again, Jeff.
It starts with `UInt256 hash = Blockchain.Singleton.CurrentBlockHash;`